contracts/contracts/strategies/balancer/BalancerMetaPoolStrategy.sol

```diff
@@ -0,0 +1,477 @@
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+contract BalancerMetaPoolStrategy is BaseAuraStrategy {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        BaseBalancerConfig memory _balancerConfig,
+        address _auraRewardPoolAddress
+    )
+        InitializableAbstractStrategy(_stratConfig)
+        BaseBalancerStrategy(_balancerConfig)
+        BaseAuraStrategy(_auraRewardPoolAddress)
+    {}
+
+    /**
+     * @notice Deposits an `_amount` of vault collateral assets
+     * from the this strategy contract to the Balancer pool.
+     * @param _asset Address of the Vault collateral asset
+     * @param _amount The amount of Vault collateral assets to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        whenNotInVaultContext
+        onlyVault
+        nonReentrant
+    {
+        address[] memory assets = new address[](1);
+        uint256[] memory amounts = new uint256[](1);
+        assets[0] = _asset;
+        amounts[0] = _amount;
+
+        _deposit(assets, amounts);
+    }
+
+    /**
+     * @notice Deposits specified vault collateral assets
+     * from the this strategy contract to the Balancer pool.
+     * @param _assets Address of the Vault collateral assets
+     * @param _amounts The amount of each asset to deposit
+     */
+    function deposit(address[] memory _assets, uint256[] memory _amounts)
+        external
+        whenNotInVaultContext
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_assets, _amounts);
+    }
+
+    /**
+     * @notice Deposits all supported assets in this strategy contract to the Balancer pool.
+     */
+    function depositAll()
+        external
+        override
+        whenNotInVaultContext
+        onlyVault
+        nonReentrant
+    {
+        uint256 assetsLength = assetsMapped.length;
+        address[] memory assets = new address[](assetsLength);
+        uint256[] memory amounts = new uint256[](assetsLength);
+
+        // For each vault collateral asset
+        for (uint256 i = 0; i < assetsLength; ++i) {
+            assets[i] = assetsMapped[i];
+            // Get the asset balance in this strategy contract
+            amounts[i] = IERC20(assets[i]).balanceOf(address(this));
+        }
+        _deposit(assets, amounts);
+    }
+
+    function _deposit(address[] memory _assets, uint256[] memory _amounts)
+        internal
+    {
+        (IERC20[] memory tokens, , ) = balancerVault.getPoolTokens(
+            balancerPoolId
+        );
+
+        uint256[] memory mappedAmounts = new uint256[](tokens.length);
+        address[] memory mappedAssets = new address[](tokens.length);
+        for (uint256 i = 0; i < _assets.length; ++i) {
```

Comment:
🟧 This method scares me because there's not an obvious check here that the passed in assets are all valid supported one.

Reply:
I don't think VaultCore can be configured to call deposit with the unsupported asset. Still this sort of safety check shouldn't be very gas intensive so adding a: 
```
require(supportsAsset(asset), "Unsupported asset");
```